scanner/src/funTest/kotlin/scanners/MultipleScannersTest.kt

```diff
@@ -0,0 +1,114 @@
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.scanner.scanners
+
+import io.kotest.core.spec.style.WordSpec
+import io.kotest.matchers.shouldBe
+
+import org.ossreviewtoolkit.model.AnalyzerResult
+import org.ossreviewtoolkit.model.AnalyzerRun
+import org.ossreviewtoolkit.model.DependencyGraph
+import org.ossreviewtoolkit.model.DependencyGraphNode
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.PackageLinkage
+import org.ossreviewtoolkit.model.PackageType
+import org.ossreviewtoolkit.model.Project
+import org.ossreviewtoolkit.model.RootDependencyIndex
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.VcsType
+import org.ossreviewtoolkit.utils.test.shouldNotBeNull
+
+private const val DEFAULT_NS = "SpdxDocumentFile"
+private const val DEFAULT_VERSION = ""
+
+class MultipleScannersTest : WordSpec({
+    "Scanning a project and a package with overlapping provenance and non overlapping scanners" should {
+        val analyzerResult = createAnalyzerResult()
+
+        val dummyScanners = DummyScanner()
+        val ortResult = createScanner().copy(
+            scannerWrappers = mapOf(
+                PackageType.PROJECT to listOf(dummyScanners.copy(name = "Dummy2")),
+                PackageType.PACKAGE to listOf(dummyScanners)
+            )
+        ).scan(analyzerResult, skipExcluded = false, emptyMap())
+
+        "return scan results with non-overlapping scanners" {
+            ortResult.scanner.shouldNotBeNull {
+                val results = getScanResults(Identifier(DEFAULT_NS, "", "spdx-project", DEFAULT_VERSION))
+                results.map { it.scanner.name }.toSet() shouldBe setOf("Dummy2")
+                val results2 = getScanResults(Identifier(DEFAULT_NS, "", "spdx-dependency", DEFAULT_VERSION))
+                results2.map { it.scanner.name }.toSet() shouldBe setOf("Dummy")
+            }
+        }
+    }
+})
+
+private fun createAnalyzerResult(): OrtResult {
+    val vcsPackage = VcsInfo(
+        type = VcsType.GIT,
+        url = "https://github.com/nnobelis/spdx-dependency.git",
+        revision = "78c8a5b12ae2d0f613fff25f2d1eb746a768ece7",
+        path = ""
+    )
+
+    val pkg = Package.EMPTY.copy(
+        id = createId("spdx-dependency"),
+        vcs = vcsPackage,
+        vcsProcessed = vcsPackage.normalize()
+    )
+
+    val vcsProject = VcsInfo(
+        type = VcsType.GIT,
+        url = "https://github.com/nnobelis/spdx-project.git",
+        revision = "8fae7f2b0efa8addeadfd15041e3437d92ff6b71",
+        path = ""
+    )
+
+    val project = Project.EMPTY.copy(
+        id = createId("spdx-project"),
+        vcs = vcsProject,
+        vcsProcessed = vcsProject.normalize()
+    )
+
+    val scopes = mapOf(":spdx-project::default" to listOf(RootDependencyIndex(0, 0)))
```

Comment:
I believe that for this test it suffices to not link the package to the project at all.
So, you  could just omit any scopes and the dependency graph. Would that make sense?

Reply:
Yes, it is working !